src/lib/swim/swim.c

```diff
@@ -2242,9 +2229,26 @@ swim_delete(struct swim *swim)
 	mh_swim_table_delete(swim->members);
 	trigger_destroy(&swim->on_member_event);
 	free(swim->shuffled);
+	TRASH(swim);
 	free(swim);
 }
 
+void
+swim_delete(struct swim *swim, bool wait)
```

Comment:
Let's leave the existing code as is and instead add a new function for `__gc`:

```C
void
swim_gc(struct swim *swim)
{
	swim->free_in_handler = true;
	fiber_set_joinable(swim->event_handler, false);
	fiber_cancel(swim->event_handler);
}
```

The fiber then would do something like this:

```C
if (swim->free_in_handler) {
	swim->event_handler = NULL;
	swim_delete(swim);
}
```